### PR TITLE
Optimizations & Simplifications - Removed multiple from String.Split to Span

### PR DESCRIPTION
**Pull request by BaronGreenback**
Replaced all string.Split with Spans.
Simplified container contains code
Commented.
Nullable'd.
Added more tests.

Changed Profile.Id to Guid - in prep for DLNA plugin.

Before anyone points it out - i've had to duplicate the SplitStringExtension.cs as it cannot be accessed from model.
Looking to move this code out post DLNA plugins, so this is really only a temp step.

### CODE REVIEW

File: Emby.Dlna/PlayTo/PlayToManager.cs [hunk 1]
```diff
@@ -217,7 +217,7 @@ private async Task AddDevice(UpnpDeviceInfo info, CancellationToken cancellation
 
                 _sessionManager.ReportCapabilities(sessionInfo.Id, new ClientCapabilities
                 {
-                    PlayableMediaTypes = profile.GetSupportedMediaTypes(),
+                    PlayableMediaTypes = profile.SupportedMediaTypes.Split(','),
```
Comment (BaronGreenback) [inline Emby.Dlna/PlayTo/PlayToManager.cs:220]: Temp solution, as this is an enum in the dlna plugin

File: MediaBrowser.Model/Extensions/ContainerExtensions.cs [hunk 2]
```diff
@@ -0,0 +1,124 @@
...
+
+            if (string.IsNullOrEmpty(inputContainer))
+            {
+                return isNegativeList;
+            }
+
+            var allInputContainers = inputContainer.SpanSplit(',');
+            foreach (var container in allInputContainers)
+            {
+                var allProfileContainers = profileContainers.SpanSplit(',');
```
Comment (crobibero) [inline MediaBrowser.Model/Extensions/ContainerExtensions.cs:74]:
```suggestion
            var allProfileContainers = profileContainers.SpanSplit(',');
            foreach (var container in allInputContainers)
            {
```

No reason to split the string multiple times

File: MediaBrowser.Model/Extensions/ContainerExtensions.cs [hunk 3]
```diff
@@ -0,0 +1,124 @@
...
+
+            if (inputContainer == null)
+            {
+                return isNegativeList;
+            }
+
+            var allInputContainers = inputContainer.Split(',');
+            foreach (var container in allInputContainers)
+            {
+                var allProfileContainers = profileContainers.SpanSplit(',');
```
Comment (crobibero) [inline MediaBrowser.Model/Extensions/ContainerExtensions.cs:111]:
```suggestion
            var allProfileContainers = profileContainers.SpanSplit(',');
            foreach (var container in allInputContainers)
            {
```

File: Emby.Dlna/DlnaManager.cs [hunk 4]
```diff
@@ -426,7 +426,7 @@ public void UpdateProfile(DeviceProfile profile)
                 throw new ArgumentException("Profile is missing Name");
             }
 
-            var current = GetProfileInfosInternal().First(i => string.Equals(i.Info.Id, profile.Id, StringComparison.OrdinalIgnoreCase));
+            var current = GetProfileInfosInternal().First(i => string.Equals(i.Info.Id, profile.Id.ToString("N", CultureInfo.InvariantCulture), StringComparison.OrdinalIgnoreCase));
```
[the remark this answers is not in the document]
Comment (BaronGreenback) [reply Emby.Dlna/DlnaManager.cs:429]: It's another temp solution. 

(I've got a PR in the pending, that removes the non-dlna part of DlnaManager into a new class where everything is treated as GUID. The remaining code is in the dlna plugin.)

**Comment by Shadowghost** [comment]
Superseded by #9374

**Pull request closed without merging**